File: client/src/app/modules/tiers/services/sales.complain.problem.service.ts
import { HttpClient, HttpParams } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { AppRestService } from 'src/app/services/appRest.service';
import { SalesComplainProblem } from '../../miscellaneous/model/SalesComplainProblem';

@Injectable({
  providedIn: 'root'
})
export class SalesComplainProblemService extends AppRestService<SalesComplainProblem>{

  constructor(http: HttpClient) {
    super(http, "tiers");
  }

  getComplainProblems() {
    return this.getList(new HttpParams(), "sales-complain-problems");
  }

   addOrUpdateComplainProblem(salesComplainProblem: SalesComplainProblem) {
    return this.addOrUpdate(new HttpParams(), "sales-complain-problem", salesComplainProblem, "Enregistré", "Erreur lors de l'enregistrement");
  }
}

File: src/main/java/com/jss/osiris/modules/miscellaneous/service/SalesComplainProblemService.java
package com.jss.osiris.modules.miscellaneous.service;

import java.util.List;

import com.jss.osiris.modules.miscellaneous.model.SalesComplainProblem;

public interface SalesComplainProblemService {

        public List<SalesComplainProblem> getComplainProblems();

        public SalesComplainProblem addOrUpdateComplainProblem(SalesComplainProblem salesComplainProblem);
}
